### PR TITLE
Added to Nested AD example how to use `batched_jacobian`

### PR DESCRIPTION
**Pull request by facusapienza21**
Just following on the discussion with @avik-pal in #963. 

I am not sure about the formatting of this, I hope there is some CI in place to be sure all looks nice. Happy to make any changes @avik-pal that you think adequate.

### CODE REVIEW

File: docs/src/manual/nested_autodiff.md [hunk 1]
```diff
@@ -106,6 +106,16 @@ nothing; # hide
 That's pretty good, of course you will have some error from the finite differences
 calculation.
 
+!!! tip
+
+    Notice that in this example the Jacobian `J` consists on the full matrix of derivatives of `smodel` with respect
+    the different inputs in `x`. In order to avoid the unnecessary calculation of zero entries of the Jacobian, you can 
+    instead use [`batched_jacobian`](@ref) to parse the calculation of the Jacobian per each single input. 
+    ```@example nested_ad
+    J = batched_jacobian(smodel, AutoForwardDiff(), x)
```
Comment (avik-pal) [inline docs/src/manual/nested_autodiff.md:115]: make it into a full function like `loss_function1`. Then if you add the checks with `@assert ... # hide` those will ensure that the computation is correct and won't be exposed in the docs

Comment (facusapienza21) [reply docs/src/manual/nested_autodiff.md:115]: I will do it! But please take a look at my following comment.

**Comment by facusapienza21** [comment]
@avik-pal the current example in the documentation includes a neural network with batch normalization, reason why the Jacobian has non-trivial entries everywhere. This is quite different than the original example I posted in Discourse, where the idea is to compute the gradient with respect to the input layer multiple times. Here we can do two things: 
- Remove the Batch normalization of the first example to reproduce that behaviour, but I guess you include it for good reasons
- Make this new example different and remove the batch normalization of the chain. 

Maybe the second option makes more sense, but in that case I will add more explanation in the text. What do you think? The two options look fine for me. 

For reference, here is `J` with BatchNorm 
image
and without it
image

**Comment by avik-pal** [comment]
Ah right forgot about that. You can change that to `GroupNorm`. I put in a normalization layer to ensure that we aren't cheating with Zygote over Zygote (which will fail for any of the normalization layers)

**Comment by avik-pal** [comment]
Alternatively you can add the `smodel = ...` in that example block. currently it doesn't have it in global scope https://buildkite.com/julialang/lux-dot-jl/builds/4386#019245db-55f8-484c-ac71-b18b7a56b970/656-1623

**Comment by facusapienza21** [comment]
Thank you @avik-pal , I just made this changes, let me know what do you think.

File: docs/src/manual/nested_autodiff.md [hunk 2]
```diff
@@ -106,6 +106,53 @@ nothing; # hide
...
+ps, st = Lux.setup(StableRNG(0), model)
+x = randn(StableRNG(0), Float32, 2, 10)
+y = randn(StableRNG(11), Float32, 2, 10)
+
+function loss_function_batched(model, x, ps, st, y)
+    # Make it a stateful layer
+    smodel = StatefulLuxLayer{true}(model, ps, st)
+    ŷ = smodel(x)
+    loss_emp = sum(abs2, ŷ .- y)
+    # You can use `Zygote.jacobian` as well but ForwardDiff tends to be more efficient here
```
Comment (avik-pal) [inline docs/src/manual/nested_autodiff.md:128]:
```suggestion
    # You can use `AutoZygote()` as well but `AutoForwardDiff()` tends to be more efficient here
```

**Review by avik-pal — APPROVED** [review]
Thanks!

**Pull request merged**